### ISSUE TITLE
Date picker causing safari to lock up

### ISSUE THREAD

**Issue by ryanking1809** [jquense/react-widgets#473]
The react widget date picker is locking up my Safari Version 10.0 (12602.1.50.0.10)
It seems to work fine on browser stack but we're hearing similar issues from other users.

There's no errors in the console indicating what going on. Cleared caches, removed extensions, and it still locks up (all you see is the spinning wheel of death & have reload the tab to get response).

Is anyone else having issues / have a solution to what going on?

**Comment by jquense** [comment]
working for me as well. I haven't heard anything either :/

**Comment by jheander** [comment]
We have plenty of similar reports from users running both Safari and Internet Explorer. I have managed to reproduce the problem by putting a date picker in an area that uses either "position: fixed" or "position: absolute".

This causes the code to get stuck in a loop when you try to open the calendar in the date picker. The loop keeps coming back to these lines in Popup.js:126

```
var height = this.height();
if (height !== this.state.height) this.setState({ height: height });
```

I added log prints here, and it seems that the calculated height and the current state is oscillating between to very similar values, differing only at the third decimal. But since they are compared to absolute equality it will go on for ever. Such is the peril of comparing floating point numbers :)

I'll send a pull request that fixes the issue. Maybe it is the same issue you are seeing ryanking1809.
